### ISSUE TITLE
Fails to run after rebuild in AWS lambda 

### ISSUE THREAD

**Issue by pookaboydunc** [damirn/rapidx2j#37]
I have been using this for some time in AWS Lambda nodejs6.10 but as they are not supporting this anymore I have decided to rebuild against nodejs8.10. I can get the rebuild fine and all works locally however I have trouble running in lambda. I get the following error:

```
module initialization error: Error
at Object.Module._extensions..node (module.js:681:18)
at Module.load (module.js:565:32)
at tryModuleLoad (module.js:505:12)
at Function.Module._load (module.js:497:3)
at Module.require (module.js:596:17)
at require (internal/module.js:11:18)
at Object.<anonymous> (/var/task/node_modules/tva2json/node_modules/rapidx2j/index.js:3:16)
at Module._compile (module.js:652:30)
at Object.Module._extensions..js (module.js:663:10)
at Module.load (module.js:565:32)
```

Any idea what I am doing wrong here?

**Comment by damirn** [comment]
no idea; can you share a complete example please?

**Comment by pookaboydunc** [comment]
I'm not really doing anything strange. I have a node module that performs a conversion and it is this that depends on rapidx2j. I can test this module fine locally and all works as expected it's only when I deploy to AWS that the module initialization error occurs.

When installing:
`npm install

> rapidx2j@0.5.0 install /home/user/project/node_modules/tva2json/node_modules/rapidx2j
> node-gyp rebuild
make: Entering directory '/home/user/project/node_modules/tva2json/node_modules/rapidx2j/build
  CXX(target) Release/obj.target/rapidx2j/src/rapidx2j.o
  CXX(target) Release/obj.target/rapidx2j/src/parser.o
  SOLINK_MODULE(target) Release/obj.target/rapidx2j.node
  COPY Release/rapidx2j.node
make: Leaving directory '/home/user/project/node_modules/tva2json/node_modules/rapidx2j/build'
npm notice created a lockfile as package-lock.json. You should commit this file.

added 153 packages from 140 contributors and audited 226 packages in 8.319s
`
Require statement:
`var x2j = require('rapidx2j');`

**Comment by damirn** [comment]
I'm not familiar with AWS lambda nor do I have a way to test it. Perhaps to file a support request to AWS?

**Comment by pookaboydunc** [comment]
@damirn I managed to sort the problem. The problem is that native modules need to be built within the same linux environment that AWS lambda uses. I found this article which explains a nifty little tricky to use docker to do a build in the correct AWS lambda env:

https://www.rookout.com/3_min_hack_for_building_local_native_extensions/

I tried this against nodejs8.10 and all is working as expected. If anyone else has this problem it is super easy to fix :)
